core/src/test/scala/neotypes/FutureAsyncSpec.scala

```diff
@@ -0,0 +1,92 @@
+package neotypes
+
+import org.scalatest.{FlatSpec, Matchers}
+
+import scala.concurrent.{ExecutionContext, Future}
+import org.scalatest.concurrent.ScalaFutures
+class FutureAsyncSpec extends FlatSpec with Matchers {
+  implicit val ec = ExecutionContext.Implicits.global
+
+  trait FutureAsyncFixture {
+    var counter = 0
+    var resourceIsOpen = true
+    val fa = Future.successful(1)
+    val f: Int => Future[String] = i => Future.successful(i.toString)
+    val finalizer = { (_: Int, _: Option[Throwable]) => {Future.successful {resourceIsOpen = false}}}
+  }
+
+  "async future" should "execute finalizer" in new FutureAsyncFixture{
+    ScalaFutures.whenReady(Async.futureAsync.guarantee(fa)(f)(finalizer)){
+      w =>
```

Comment:
Could you give this a variable a better name? I do not understand what `w` is supposed to mean.